### DISCUSSION TITLE
Easy configuration Pow-K+ via Android app

### DISCUSSION THREAD

**Discussion by stahjo** [UtilitechAS/amsreader-firmware#653] [Q&A]
I'm building av visualization app to be used by utilities i Sweden. In this app there will be an easy installation wizard so the app automatically connect the Pow-K+ to the predefined MQTT broker when the user have defined which local WiFi network the Pow-K+ shall connect to. Is it possible to set the configuration parameters via http (or any other Set command) directly from the app and skip the configuration on the local webpages on the Pow-K+?
I'm using the Pow-K+ with firmware v2.2.22.

**Comment by gskjold** [comment]
This is absolutely possible. All config change requests should be sent to /save. 

For initial setup, send these parameters:
s=true // Enable setup config
ss=SSID
sp=PSK
sm=dhcp/static ("enum")
si=static_ip
sg=static_gw
su=static_subnet
sd=static_dns
sh=hostname

We are also looking into smart config (See #609)

After device has connected to WiFi, use the following for MQTT:
q=true // Enable MQTT config
qh=host
qc=client_id
qb=publis_topic
qu=username
qa=pass
qm=0,1,2,3 or 4 (0=json, 1=raw_min, 2=raw_full, 3=domoticz, 4=ha)
qs=SSL
qp=port

Also, in 2.2.23 (next patch), MQTT SSL have been improved a little bit, allowing to connect to broker without uploading CA, cert and key.
